src/client/dfs/duns.c

```diff
@@ -697,44 +780,7 @@ duns_create_path(daos_handle_t poh, const char *path, struct duns_attr_t *attrp)
 			goto err_link;
 		}
 
-		if (attrp->da_type == DAOS_PROP_CO_LAYOUT_POSIX) {
-			dfs_attr_t dfs_attr = {};
-
-			/** TODO: set Lustre FID here. */
-			dfs_attr.da_id = 0;
-			dfs_attr.da_oclass_id = attrp->da_oclass_id;
-			dfs_attr.da_chunk_size = attrp->da_chunk_size;
-			dfs_attr.da_props = attrp->da_props;
-			rc = dfs_cont_create(poh, attrp->da_cuuid, &dfs_attr,
-					     NULL, NULL);
-		} else {
-			daos_prop_t	*prop;
-			int		 nr = 1;
-
-			if (attrp->da_props != NULL)
-				nr = attrp->da_props->dpp_nr + 1;
-
-			prop = daos_prop_alloc(nr);
-			if (prop == NULL) {
-				D_ERROR("Failed to allocate container prop.");
-				D_GOTO(err_link, rc = ENOMEM);
-			}
-			if (attrp->da_props != NULL) {
-				rc = daos_prop_copy(prop, attrp->da_props);
-				if (rc) {
-					daos_prop_free(prop);
-					D_ERROR("failed to copy properties (%d)\n", rc);
-					return daos_der2errno(rc);
-				}
-			}
-			prop->dpp_entries[prop->dpp_nr - 1].dpe_type =
-				DAOS_PROP_CO_LAYOUT_TYPE;
-			prop->dpp_entries[prop->dpp_nr - 1].dpe_val =
-				attrp->da_type;
-			rc = daos_cont_create(poh, attrp->da_cuuid, prop, NULL);
-			daos_prop_free(prop);
-		}
-
+		rc = create_cont(poh, attrp);
 		if (rc == -DER_SUCCESS && backend_dfuse) {
```

Comment:
Since `create_cont` returns a `daos_der2errno`, should we still check `-DER_SUCCESS`? But I guess that's still 0.

Reply:
yea. anyway just FYI this isn't anything that was changed in this ticket.

Reply:
Yeah, no worries.